addons/l10n_my_edi_extended/models/account_edi_xml_ubl_my.py

```diff
@@ -20,10 +20,25 @@ def _export_invoice_vals(self, invoice):
                 'industry_classification_code_attrs': {'name': invoice.partner_id.commercial_partner_id.l10n_my_edi_industrial_classification.name},
                 'industry_classification_code': invoice.partner_id.commercial_partner_id.l10n_my_edi_industrial_classification.code,
             })
+        # Sometimes, a foreign customer is also a supplier.
+        # To avoid needing to change the Generic TIN depending on what you do with your commercial partner, we will automatically switch
+        # depending on the context.
+        if vals['vals']['document_type_code'] in ('11', '12', '13'):
+            other_party = vals['vals']['accounting_supplier_party_vals']['party_vals']
+            # Switch from foreign buyer to foreign supplier
+            for identification_val in other_party['party_identification_vals']:
+                if identification_val.get('id_attrs', {}).get('schemeID') == 'TIN' and identification_val.get('id') == 'EI00000000020':
+                    identification_val['id'] = 'EI00000000030'
+        else:
+            other_party = vals['vals']['accounting_customer_party_vals']['party_vals']
+            # Switch from foreign supplier to foreign buyer
+            for identification_val in other_party['party_identification_vals']:
+                if identification_val.get('id_attrs', {}).get('schemeID') == 'TIN' and identification_val.get('id') == 'EI00000000030':
+                    identification_val['id'] = 'EI00000000020'
```

Comment:
Maybe it would make sense to move this logic to `_get_partner_party_vals`?

And what do you think about pulling the logic / loop out of the `if`/`else`?
```suggestion
        if vals['vals']['document_type_code'] in ('11', '12', '13'):
            other_party = vals['vals']['accounting_supplier_party_vals']['party_vals']
            our_role_foreign_TIN = 'EI00000000020'  # generic foreign customer
            other_role_foreign_TIN = 'EI00000000030'  # generic foreign supplier
        else:
            other_party = vals['vals']['accounting_customer_party_vals']['party_vals']
            our_role_foreign_TIN = 'EI00000000030'
            other_role_foreign_TIN = 'EI00000000020'
        # Adapt generic foreign TINs for the other party to their role (buyer / seller)
        for identification_val in other_party['party_identification_vals']:
            if identification_val.get('id_attrs', {}).get('schemeID') == 'TIN' and identification_val.get('id') == our_role_foreign_TIN:
                identification_val['id'] = other_role_foreign_TIN
```